### ISSUE TITLE
Question about the pretraining.

### ISSUE THREAD

**Issue by Yangr116** [Stability-AI/generative-models#86]
Hi, I would like to finetune SDXL on the custom dataset. I merge the inference config with the training config in order to load the pretrained model. However, there is an error about ```target_size_as_tuple```. 

Where should I put the ```target_hight``` and ```target_width``` in my own dataset?

**Comment by iamkrish001** [comment]
from transformers import SDXLTokenizer, SDXLForConditionalGeneration, Trainer, TrainingArguments, DataCollatorForSeq2Seq

# Load tokenizer and model
tokenizer = SDXLTokenizer.from_pretrained("path_to_pretrained_model")
model = SDXLForConditionalGeneration.from_pretrained("path_to_pretrained_model")

# Prepare your custom dataset
train_dataset = ...  # Your custom dataset

# Define the data collator for Seq2Seq task
data_collator = DataCollatorForSeq2Seq(tokenizer=tokenizer)

# Training arguments
training_args = TrainingArguments(
    output_dir="./output_dir",
    num_train_epochs=3,
    per_device_train_batch_size=2,
    save_steps=500,
    save_total_limit=2,
)

# Create Trainer and fine-tune the model
trainer = Trainer(
    model=model,
    args=training_args,
    data_collator=data_collator,
    train_dataset=train_dataset,
)

trainer.train()

**Comment by Yangr116** [comment]
Thanks, I will try this.

**Comment by knoopx** [comment]
> from transformers import SDXLTokenizer, SDXLForConditionalGeneration, Trainer, TrainingArguments, DataCollatorForSeq2Seq
> 
> # Load tokenizer and model
> tokenizer = SDXLTokenizer.from_pretrained("path_to_pretrained_model") model = SDXLForConditionalGeneration.from_pretrained("path_to_pretrained_model")
> 
> # Prepare your custom dataset
> train_dataset = ... # Your custom dataset
> 
> # Define the data collator for Seq2Seq task
> data_collator = DataCollatorForSeq2Seq(tokenizer=tokenizer)
> 
> # Training arguments
> training_args = TrainingArguments( output_dir="./output_dir", num_train_epochs=3, per_device_train_batch_size=2, save_steps=500, save_total_limit=2, )
> 
> # Create Trainer and fine-tune the model
> trainer = Trainer( model=model, args=training_args, data_collator=data_collator, train_dataset=train_dataset, )
> 
> trainer.train()

is this an LLM output? neither SDXLTokenizer or SDXLForConditionalGeneration exist.

**Comment by RoganInglis** [comment]
I have got it working by passing these 3 additional values in the input batch when using a custom dataset:
```
original_size_as_tuple - dtype: int64, shape: (batch, 2), the original size (h, w) of the image before cropping
crop_coords_top_left - dtype: int64, shape: (batch, 2), the coords (top, left) of the crop box when cropping the input to square
target_size_as_tuple - dtype: int64, shape: (batch, 2), the size (h, w) of the image after cropping to square
```

**Comment by RoganInglis** [comment]
I've only been fine-tuning the conditioning network so far but I have managed to get things into 16 bit by adding 
`precision: '16-mixed'` to the trainer section of the config. I found `'bf16-mixed'` doesn't work with PyTorch 2.0 but might work with 2.1 due to a bug fix. I haven't confirmed that this affects the UNet but I have seen that the overall memory usage has gone down a fair bit. See [here](https://lightning.ai/docs/pytorch/stable/common/trainer.html) for the official documentation of the trainer args including 'precision'

**Comment by Yangr116** [comment]
Yes, I set the``` precision``` to ```16``` in the trainer of the config, but it fails to work. I will try the Pytorch 2.0. Thanks.
